### ISSUE TITLE
setup.sh does not correctly extract GoogleSignIn

### ISSUE THREAD

**Issue by chrisellsworth** [firebase/FirebaseUI-iOS#27]
I'm unable to build FirebaseUI.xcodeproj (a45202b) because the `GoogleSignIn` import fails:

```
/.../FirebaseUI-iOS/FirebaseUI/Auth/API/FirebaseLoginViewController.h:39:9: fatal error:
      'GoogleSignIn/GoogleSignIn.h' file not found
#import <GoogleSignIn/GoogleSignIn.h>
        ^


** BUILD FAILED **
```

This seems to be due to a mismatch between downloading 2.4.0 and expecting 2.2.0. I can change the `GOOGLE_SDK_DIR` in `setup.sh` to point to `"$SCRIPT_DIR/sdk/google_signin_sdk_2_4_0"` and fix the corresponding entry in the framework search paths to make the build succeed, but I haven't tried to run anything against it yet so this may not be sufficient.

---

Furthermore, I see there are still a couple other errors when I run `setup.sh`, and though they don't seem to prevent the build from completing, I'll mention them here in case they are of importance. Namely:

```
./setup.sh: line 88: [-d: command not found
```

and

```
mv: rename /.../FirebaseUI-iOS/sdk/google-core-sdk.zip/* to /.../FirebaseUI-iOS/sdk/*: Not a directory
```

For the first error, my machine (running El Capitan, bash `3.2.57(1)-release`, nothing fancy) doesn't seem to like the `[-d` syntax so I need to change this line:

```
if [-d "$GOOGLE_CORE_SDK_DIR"]; then
```

to this:

```
if [ -d "$GOOGLE_CORE_SDK_DIR" ]; then
```

The second error appears to be related to these values being the same:

```
GOOGLE_CORE_SDK_ZIP_FILE="$SDK_DIR/google-core-sdk.zip"
GOOGLE_CORE_SDK_DIR="$SCRIPT_DIR/sdk/google-core-sdk.zip"
```

Then I imagine we'd want to copy everything from `sdk/google-core-sdk/Libraries` into `sdk/`

**Comment by pmodernme** [comment]
I am having the same issue

**Comment by mcdonamp** [comment]
Yeah, I can see the issue with 2.4.0--I just did that spur of the moment from 2.2.0 and it must be different. As for the other errors: @davideast may have some more ideas as he wrote the original version of this.

**Comment by mcdonamp** [comment]
Fix is in https://github.com/firebase/FirebaseUI-iOS/commit/e6edbb48b74522546d4a8dba04bc598dbfbb74dd

**Comment by mcdonamp** [comment]
Fixed in #28
